Homework_Collection/Homework_Collection/Stockroom.cs

```diff
@@ -0,0 +1,71 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+
+namespace Homework_Collection
+{
+    internal class Stockroom<TmyType> : IEnumerable<TmyType>, IEnumerator<TmyType>
+    {
+        private LinkedList<TmyType> stuff = new LinkedList<TmyType>();
+        private LinkedListNode<TmyType> stuffNode;
+        
+        public TmyType Current
+        {
+            get
+            {
+                return stuffNode.Value;
+            }
+        }
+
+        object IEnumerator.Current => Current;
+
+        public void Dispose()
+        {
+
+        }
+
+        public IEnumerator GetEnumerator()
+        {
+            stuffNode = null;
+            return this;
+        }
+
+        public bool MoveNext()
+        {
+            if (stuffNode == null)
+            {
+                stuffNode = stuff.First;
+            }
+            else { stuffNode = stuffNode.Next; }
+
+            return stuffNode != null;
+        }
+
+        public void Reset()
+        {
+            throw new NotImplementedException();
+        }
+
+        public void Add(TmyType stockItem)
+        {
+             stuff.AddLast(stockItem);
+        }
+
+        public void RemoveAt(int position)
+        {
+            stuffNode = stuff.First;
+
+            for (int i = 0; i < position-1; i++)
+            {
+                stuffNode = stuffNode.Next;
+            }
+
+            stuff.Remove(stuffNode);
+        }
+
+       IEnumerator<TmyType> IEnumerable<TmyType>.GetEnumerator()
+        {
+            throw new NotImplementedException();
```

Comment:
Коментарій звідси https://stackoverflow.com/questions/14909326/when-does-ienumerable-getenumerator-get-called-instead-of-ienumerablet-getenum

The non-generic IEnumerator will be executed whenever code casts the class to the non-generic interface:

((IEnumerable)myWords).GetEnumerator(); // this calls the non-generic one
This is mostly relevant if you pass your class to some legacy function that requires a non-generic IEnumerator.

So if you have some library that contains a function and you pass your class to this function, it will use the non-generic IEnumerator

DoSomeStuffWithAnIEnumerable(IEnumerable x)
{
var foo = x.GetEnumerator();

// or, as stackx said, an example with foreach:
foreach (var foo2 in x)
Console.WriteLine(foo2);
}

DoSomeStuffWithAnIEnumerable(new myWords());

Note that it is perfectly valid to simply implement the non-generic IEnumerator using the generic one:

public class myWords : IEnumerable
{
....
IEnumerator IEnumerable.GetEnumerator()
{
return this.GetEnumerator();
}
}
That way you can be sure that they both have the same effects.